src/atlas/array/helpers/ArrayForEach.h

```diff
@@ -245,89 +272,89 @@ struct ArrayForEach {
   ///         and is executed with signature g(idx_i, idx_j,...), where the idxs
   ///         are indices of ItrDims.
   ///         When a config is supplied containing "execution_policy" =
-  ///         "parallel_unsequenced_policy" (default) the first loop is executed
-  ///         using OpenMP. The remaining loops are executed in serial. When
-  ///         "execution_policy" = "sequenced_policy", all loops are executed in
-  ///         sequential (row-major) order.
+  ///         "sequenced_policy" (default). All loops are then executed in sequential
+  ///         (row-major) order.
+  ///         With "execution_policy" = "parallel_unsequenced" the first loop is executed
+  ///         using OpenMP. The remaining loops are executed in serial.
   ///         Note: The lowest ArrayView.rank() must be greater than or equal
   ///         to the highest dim in ItrDims. TODO: static checking for this.
-  template <typename ArrayViewTuple, typename Mask, typename Function,
-            typename = std::enable_if_t<detail::is_tuple<ArrayViewTuple>()>>
+  template <typename... ArrayView, typename Mask, typename Function>
   static void apply(const eckit::Parametrisation& conf,
-                    ArrayViewTuple&& arrayViews,
+                    std::tuple<ArrayView...>&& arrayViews,
                     const Mask& mask, const Function& function) {
 
     auto execute = [&](auto execution_policy) {
-      apply(execution_policy, std::forward<ArrayViewTuple>(arrayViews), mask, function);
+      apply(execution_policy, std::move(arrayViews), mask, function);
     };
 
     using namespace execution;
     std::string execution_policy;
-    if( conf.get("execution_policy",execution_policy) ) {
+    if (conf.get("execution_policy",execution_policy)) {
       if (execution_policy == policy_name(par_unseq)) {
         execute(par_unseq);
-      } else if (execution_policy == policy_name(par)) {
+      }
+      else if (execution_policy == policy_name(par)) {
         execute(par);
-      } else if (execution_policy == policy_name(unseq)) {
+      }
+      else if (execution_policy == policy_name(unseq)) {
         execute(unseq);
-      } else if (execution_policy == policy_name(seq)) {
+      }
+      else if (execution_policy == policy_name(seq)) {
         execute(seq);
       }
+      else {
+        throw_Exception("Unrecognized execution policy "+execution_policy, Here());
+      }
     }
     else {
-      execute(par_unseq);
+      execute(seq);
```

Comment:
I agree, sequenced default makes sense.

Possible oversight on my behalf, but is it worth changing line 116 to: ` if constexpr(std::is_same_v<Policy, execution::parallel_policy> || std::is_same_v<Policy, execution::parallel_unsequenced_policy>)` ?

I think OpenMP parallel threads support messy things like `new` and `delete`, and are therefore better described by [`par`](https://en.cppreference.com/w/cpp/algorithm/execution_policy_tag_t).

Reply:
Reading up a bit further, It would make sense to disable the `apply` interfaces which have a mask argument when the policy is `unseq` or `par_unseq`. The presence of a mask will almost certainly ruin the chances of vectorisation.

Reply:
Not 100% sure if we cannot have a mask with unseq though. Compilers can sometimes hoist this thing out for vectorisation. To be further investigated, better another time.

Reply:
I implemented the NoMask constexpr struct. It already avoids some std::make_tuple generation of maskArgs, so possible performance gain, who knows.

Reply:
> Not 100% sure if we cannot have a mask with unseq though. Compilers can sometimes hoist this thing out for vectorisation. To be further investigated, better another time.

On reflection, the chances that I'm overthinking this are high!